src/flask_container_scaffold/app_scaffold.py

```diff
@@ -1,98 +1,18 @@
-        # the ones we already have, as they are needed for the current code to
-        # work.
-        Flask.jinja_options = dict(Flask.jinja_options, trim_blocks=True,
-                                   lstrip_blocks=True)
-        self.app = (app or
-                    Flask(name,
-                          instance_relative_config=instance_relative_config,
-                          instance_path=instance_path))
-        self.config = config
-        self.silent = not settings_required
-        self.relative = instance_relative_config
-        self._init_app()
-
-    def _init_app(self):
-        self._load_flask_settings()
-        self._load_custom_settings()
-
-    def _load_flask_settings(self):
-        """
-        This loads the 'core' settings, ie, anything you could set directly
-        on a Flask app. These can be specified in the following order, each
-        overriding the last, if specified:
-        - via config mapping
-        - via Flask settings.cfg file
-        - via environment variable 'FLASK_SETTINGS'
-        """
-        config_not_loaded = True
-        if self.config is not None:
-            # load the config if passed in
-            self.app.config.from_mapping(self.config)
-            config_not_loaded = False
-        # load the instance config, if it exists and/or is required
-        try:
-            self.app.config.from_pyfile('settings.cfg', silent=self.silent)
-            config_not_loaded = False
-        except Exception:
-            config_not_loaded = True
-        # Load any additional config specified in the FLASK_SETTINGS file,
-        # if it exists. We only want to fail in the case where settings are
-        # required by the app.
-        if ((config_not_loaded and not self.silent) or
-                os.environ.get('FLASK_SETTINGS')):
-            self.app.config.from_envvar('FLASK_SETTINGS')
-
-    def _load_custom_settings(self):
-        """
-        Load any custom configuration for the app from:
-        - app.config['CUSTOM_SETTINGS']
-        - environment variable 'CUSTOM_SETTINGS'
+        This class provides compatibility with versions of scaffold that
+        expect an instance with an 'app' attribute. All of the parameters are
+        the same as BaseScaffold and are passed directly through unmodified.
         """
-        configurator = AppConfigurator(self.app, self.relative)
-        if self.app.config.get('CUSTOM_SETTINGS') is not None:
-            # load the config if passed in
-            custom = self.app.config.get('CUSTOM_SETTINGS')
-            configurator.parse(custom)
-        # Next, load from override file, if specified
-        if os.environ.get('CUSTOM_SETTINGS') is not None:
-            custom = os.environ.get('CUSTOM_SETTINGS')
-            configurator.parse(custom)
+        super().__init__(app, name, config, settings_required,
+                         instance_path, instance_relative_config)
+        self.app = app or self.flask_app
```

Comment:
I think keeping this as a separate class is good. I like that it is explicitly a different path than a CeleryScaffold. One question though. Since CeleryScaffold descends from BaseScaffold, is there a case where this class would receive a flask_app?

Reply:
I guess if someone explicitly inits with app=None, self.app would end up referring back to and using the new Flask() created in the BaseScaffold?

Reply:
Correct, the BaseScaffold will generate a Flask instance. If it's helpful, I can add a unit test for that case.

Reply:
I think we can treat that as an enhancement for later, this looks good to me!